From 4013b1be5ea4479131f2a15d7acad9dba90ae28e Mon Sep 17 00:00:00 2001
From: Liang Li <liang.li@windriver.com>
Date: Thu, 1 Sep 2011 19:59:25 +0800
Subject: [PATCH] wrhv: core: print warn message when vbi_find_irq fails

CQID: WIND00289961 (for SQA folks, :))

This commit should be folded into 'wrhv: core' commit, then
update this patch:

wrs/cfg/kernel-cache/features/multicore/wrhv-core-introduce-the-core-WR-hypervisor-support.patch

Signed-off-by: Liang Li <liang.li@windriver.com>
---
 kernel/vbi/wrhv.c |   11 +++++++----
 1 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/kernel/vbi/wrhv.c b/kernel/vbi/wrhv.c
index a4bdad9..880ed63 100644
--- a/kernel/vbi/wrhv.c
+++ b/kernel/vbi/wrhv.c
@@ -479,9 +479,14 @@ static void __devinit pci_fixup_wrhv(struct pci_dev *dev)
 				sharedint);
 		}
 		irq = vbi_find_irq(devname, 1);
-		if (irq == VBI_INVALID_IRQ)
+		if (irq == VBI_INVALID_IRQ) {
 			irq = WRHV_POLL_IRQ;
-		else
+			printk(KERN_WARNING "WRHV-PCI: find IRQ for %04x:%02x:%02x.%d "
+				   "by name [%s] failed. You are now in polling mode.\n",
+				   pci_domain_nr(dev->bus),dev->bus->number,
+				   PCI_SLOT(dev->devfn), PCI_FUNC(dev->devfn),
+				   devname);
+		} else
 			printk(KERN_INFO "WRHV-PCI: %04x:%02x:%02x.%d shared irq [%d]", pci_domain_nr(dev->bus),
 				dev->bus->number, PCI_SLOT(dev->devfn), PCI_FUNC(dev->devfn), irq);
 
@@ -636,8 +641,6 @@ int __weak wrhv_arch_late_init(void)
 	return 0;
 }
 
-#define WINDRIVER_NAME "windriver"
-
 /* sysfs ksets */
 struct kset *windriver_kset;
 EXPORT_SYMBOL(windriver_kset);
-- 
1.7.0.4

